simplifier/world/nodes.py

```diff
@@ -114,6 +114,15 @@ def simplify(self):
         for node in list(self.world.iter_postorder(self)):
             if isinstance(node, BitwiseNegate):
                 node.dissolve_negation()
+        from simplifier.range_simplifier import RangeSimplifier
+
+        for node in list(self.world.iter_postorder(self)):
+            if isinstance(node, Operation):
+                RangeSimplifier.simplify(node)
+                from simplifier.operations.interface import AssociativeOperation, CommutativeOperation
+
+                if isinstance(node, (AssociativeOperation, CommutativeOperation)):
+                    node._promote_single_operand()
         for node in list(self.world.iter_postorder(self)):
             if isinstance(node, Operation):
                 node.simplify()
```

Comment:
I would prefer to remove these lines here and would like to change line https://github.com/fkie-cad/dewolf-logic/blob/37758b53152ccf678a84c948921e0bba814686e5/simplifier/range_simplifier.py#L193 into the following:
`if isinstance(self._world.get_definition(self._defining_variable), Operation)`. 
I think this way it could be possible to also remove lines 124+125 (But here I am not sure).

But if you prefer it this way it is all right. I only thought that iterating three times the set of nodes it quite heavy.

Reply:
You're right, I've made the changes and did some testing and it still seems to work fine. So hopefully it should be ok!